Frends.MicrosoftSQL.ExecuteQuery/Frends.MicrosoftSQL.ExecuteQuery.Tests/UnitTests.cs

```diff
@@ -0,0 +1,3708 @@
+        GO
+   */
+
+    private static readonly string _connString = "Server=127.0.0.1,1433;Database=Master;User Id=SA;Password=Salakala123!";
+    private static readonly string _tableName = "TestTable";
+    
+    [TestInitialize]
+    public void Init()
+    {
+        using var connection = new SqlConnection(_connString);
+        connection.Open();
+        var createTable = connection.CreateCommand();
+        createTable.CommandText = $@"IF NOT EXISTS (SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME='{_tableName}') BEGIN CREATE TABLE {_tableName} ( Id int, LastName varchar(255), FirstName varchar(255) ); END";
+        createTable.ExecuteNonQuery();
+        connection.Close();
+        connection.Dispose();
+    }
+
+    [TestCleanup]
+    public void CleanUp()
+    {
+        using var connection = new SqlConnection(_connString);
+        connection.Open();
+        var createTable = connection.CreateCommand();
+        createTable.CommandText = $@"IF EXISTS (SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME='{_tableName}') BEGIN DROP TABLE IF EXISTS {_tableName}; END";
+        createTable.ExecuteNonQuery();
+        connection.Close();
+        connection.Dispose();
+    }
+
+    [TestMethod]
+    [ExpectedException(typeof(Exception))]
+    public async Task TestExecuteQuery_Invalid_Creds_ThrowError()
+    {
+        var input = new Input()
+        {
+            ConnectionString = "Server=127.0.0.1,1433;Database=Master;User Id=SA;Password=WrongPassWord",
+        };
+
+        var options = new Options()
+        {
+            SqlTransactionIsolationLevel = SqlTransactionIsolationLevel.ReadCommitted,
+            CommandTimeoutSeconds = 2,
+            ThrowErrorOnFailure = true
+        };
+
+        await MicrosoftSQL.ExecuteQuery(input, options, default);
+    }
+
+    [TestMethod]
+    public async Task TestExecuteQuery_Invalid_Creds_ReturnErrorMessage()
+    {
+        var input = new Input()
+        {
+            ConnectionString = "Server=127.0.0.1,1433;Database=Master;User Id=SA;Password=WrongPassWord",
+        };
+
+        var options = new Options()
+        {
+            SqlTransactionIsolationLevel = SqlTransactionIsolationLevel.ReadCommitted,
+            CommandTimeoutSeconds = 2,
+            ThrowErrorOnFailure = false
+        };
+
+        var result = await MicrosoftSQL.ExecuteQuery(input, options, default);
+        Assert.IsFalse(result.Success);
+        Assert.IsTrue(result.ErrorMessage.Contains("Login failed for user 'SA'."));
+        Assert.AreEqual(0, result.RecordsAffected);
+    }
+
+    #region Rollback
+    /*
+    // Rollback test. Commented because this requires manual modification and can't be ran on workflow.
+    // Add following line 'throw new Exception();' under 'dataReader = await command.ExecuteReaderAsync(cancellationToken).ConfigureAwait(false);'
+    [TestMethod]
+    public async Task TestExecuteQuery_RollbackInsert_ThrowErrorOnFailure_False()
```

Comment:
Instead of commenting all of this out I think it would be a better option to create a new test file named `ManualTesting` and then move these tests there. You can then add `[ignore("Only run manually")]` so CI will ignore these tests.